test/DictVectors.jl

```diff
@@ -0,0 +1,335 @@
+            @test_broken missing * dvec ≡ dvec * missing ≡ missing
+        end
+        @testset "add!" begin
+            dvec1 = type(Dict(pairs), cap)
+            dvec2 = type(Dict(pairs[1:2:end]), cap)
+            add!(dvec1, dvec2)
+            for (i, (k, v)) in enumerate(pairs)
+                if isodd(i)
+                    @test dvec1[k] == 2v
+                else
+                    @test dvec1[k] == v
+                end
+            end
+
+            copy!(dvec2, dvec1)
+            add!(dvec1, type{K,V}(cap))
+            @test dvec1 == dvec2
+        end
+        @testset "axpy!" begin
+            dvec1 = type(Dict(pairs), cap)
+            dvec2 = type(Dict(pairs[1:2:end]), cap)
+            axpy!(V(2), dvec1, dvec2)
+            for (i, (k, v)) in enumerate(pairs)
+                if isodd(i)
+                    @test dvec2[k] == 3v
+                else
+                    @test dvec2[k] == 2v
+                end
+            end
+
+            ys = Tuple(empty(dvec1) for i in 1:Threads.nthreads())
+            axpy!(2.0, dvec1, ys; batchsize=100)
+        end
+        @testset "axpby!" begin
+            dvec1 = type(Dict(pairs), cap)
+            dvec2 = type(Dict(pairs[1:2:end]), cap)
+            axpby!(V(2), dvec1, V(3), dvec2)
+            for (i, (k, v)) in enumerate(pairs)
+                if isodd(i)
+                    @test dvec2[k] == 5v
+                else
+                    @test dvec2[k] == 2v
+                end
+            end
+        end
+        @testset "dot" begin
+            dvec1 = type(Dict(pairs), cap)
+            dvec2 = type(Dict(pairs[1:2:end]), cap)
+            dvec3 = type(Dict(pairs[2:2:end]), cap)
+
+            @test dvec1 ⋅ dvec1 ≈ norm(dvec1)^2
+            @test dvec1 ⋅ dvec2 ≈ norm(dvec2)^2
+            @test dot(dvec1, (dvec2, dvec3)) ≈ norm(dvec1)^2
+        end
+        @testset "iteration" begin
+            dvec = type(Dict(pairs), cap)
+
+            dvec_pairs = [kv for kv in Base.pairs(dvec)]
+            @test issetequal(pairs, dvec_pairs)
+
+            dvec_keys = [k for k in Base.keys(dvec)]
+            @test issetequal(dvec_keys, keys)
+
+            dvec_values = [kv for kv in dvec]
+            @test issetequal(values, dvec_values)
+            dvec_values = [k for k in Base.values(dvec)]
+            @test issetequal(dvec_values, values)
+        end
+        @testset "projection" begin
+            dvec = type(Dict(pairs), cap)
+            @test UniformProjector() ⋅ dvec == sum(dvec)
+            @test NormProjector() ⋅ dvec == norm(dvec, 1)
+            @test Norm2Projector() ⋅ dvec == norm(dvec, 2)
+        end
+    end
+end
+
+@testset "DVec" begin
+    keys1 = shuffle(1:10)
+    vals1 = shuffle(1:10) .* rand((-1.0, 1.0), 10)
+    test_dvec_interface(DVec, keys1, vals1, 10)
+
+    keys2 = ['x', 'y', 'z', 'w', 'v']
+    vals2 = [1.0 + 2.0im, 3.0 - 4.0im, 0.0 - 5.0im, -2.0 + 0.0im, 12.0 + im]
+    test_dvec_interface(DVec, keys2, vals2, 100)
+end
+
```

Comment:
When I run the file on my computer I get

```
Test Summary: | Pass  Broken  Total
DVec          |  258       2    260
Test Summary: | Pass  Total
FastDVec      |   15     15
Test Summary: | Pass  Total
DFVec         |   19     19
```
Is there a way to find out which of the tests are broken and why? Should we worry, and try to fix the broken tests?

Reply:
Broken tests are a way to tell Julia you know something is failing, but you don't want it to fail your entire test suite. They are prefixed with `@test_broken`.

In this case, that's the one that tries multiplying by `missing` from both sides.

Reply:
I notice that the tests also report being broken on the GitHub action (after pull request).

Reply:
OK, I get it now. This is about multiplying `missing` from the right. All is good.